### PR TITLE
SNOW-683060 alter session to set timezone to fix failing test 

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-683060

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   To fix test failures when run in pre-commit jenkins environment.

### CODE REVIEW

File: tests/integ/scala/test_function_suite.py [hunk 1]
```diff
@@ -2229,6 +2229,7 @@ def test_as_object(session):
 
 
 def test_timestamp_tz_from_parts(session):
+    session.sql("alter account set timezone = 'America/Los_Angeles'").collect()
```
Comment (sfc-gh-sfan) [inline tests/integ/scala/test_function_suite.py:2232]: This probably won't work in stored proc, or should not be used for remote test. What is the failure that we are trying to solve?

**Review by sfc-gh-sfan — CHANGES_REQUESTED** [review]

File: tests/integ/scala/test_function_suite.py (hunk 1 above)
Comment (sfc-gh-aalam) [reply tests/integ/scala/test_function_suite.py:2232]: is there anything else we could do here?
a sample failed test: https://ci-dev-133.int.snowflakecomputing.com/job/RT-PC-TestWorker/29487/testReport/junit/(root)/t_python_sp/snowpark_scala_test_function_suite_sql/

Comment (sfc-gh-sfan) [reply tests/integ/scala/test_function_suite.py:2232]: You could add an `alter session` in the regression test itself :)

Comment (sfc-gh-aalam) [reply tests/integ/scala/test_function_suite.py]: oh yeah, that makes more sense

File: tests/integ/scala/test_function_suite.py [hunk 2]
```diff
@@ -2229,6 +2229,7 @@ def test_as_object(session):
 
 
 def test_timestamp_tz_from_parts(session):
+    session.sql("alter session set timezone = 'America/Los_Angeles'").collect()
```
Comment (sfc-gh-jdu) [inline tests/integ/scala/test_function_suite.py:2232]: Is it possible to set it with statement params? https://github.com/snowflakedb/snowpark-python/blob/7c6336bc2ee1761e1fad23da2432e7a131becaa9/src/snowflake/snowpark/dataframe.py#L535

If you change the parameter here, the following tests will also be impacted.

**Review by sfc-gh-jdu — APPROVED** [review]

**Pull request closed without merging**